### PR TITLE
CASM-2670: Refactor to use patched CSM images

### PR DESCRIPTION
**Pull request by wiltonpaulo**
CASM-2670: Refactor to use patched CSM images

### CODE REVIEW

**Comment by zcrisler** [comment]
Job `lint-test-scan / scan-images (artifactory.algol60.net/csm-docker/stable/cray-sts:0.7.0)` fails because that image does not yet exist until this PR is merged and `cray-sts-0.7.0` is released. However, here are the Snyk results from the Jenkins CI image build:

```
$ snyk container test artifactory.algol60.net/csm-docker/unstable/cray-sts:0.7.0-20211130080124_c68225b

Testing artifactory.algol60.net/csm-docker/unstable/cray-sts:0.7.0-20211130080124_c68225b...

Organization:      shasta-csm-oss
Package manager:   apk
Project name:      docker-image|artifactory.algol60.net/csm-docker/unstable/cray-sts
Docker image:      artifactory.algol60.net/csm-docker/unstable/cray-sts:0.7.0-20211130080124_c68225b
Platform:          linux/amd64
Base image:        alpine:3.15.0
Licenses:          enabled

✔ Tested 27 dependencies for known issues, no vulnerable paths found.

According to our scan, you are currently using the most secure version of the selected base image
```

File: kubernetes/cray-sts/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+home: https://github.com/Cray-HPE/cray-sts
+dependencies:
+  - name: cray-service
+    version: ~7.0.0
+    repository: https://artifactory.algol60.net/artifactory/csm-helm-charts/
+maintainers:
+  - name: kimjensen-hpe
+  - name: bklei
+  - name: brantk-hpe
+appVersion: 0.7.0
```
Comment (brantk-hpe) [inline kubernetes/cray-sts/Chart.yaml:16]: The appVersion needs to be set from the VERSION = getDockerBuildVersion(isStable: env.IS_STABLE) that was set in the Jenkinsfile.github. This is how the unstable builds reference the image that was built.

**Review by brantk-hpe — CHANGES_REQUESTED** [review]

File: Jenkinsfile.github [hunk 2]
```diff
@@ -12,37 +12,38 @@ pipeline {
...
+                    }
                     steps {
                         sh "make chart"
                     }
                 }
             }
         }
 
-        stage("Publish") {
+        stage('Publish ') {
```
Comment (kimjensen-hpe) [inline Jenkinsfile.github:42]: Is this extra space an issue?  'Publish '

File: kubernetes/cray-sts/Chart.yaml (hunk 1 above)
Comment (zcrisler) [reply kubernetes/cray-sts/Chart.yaml:16]: This is handled by the `chart-metadata` target when running `make chart`. You can verify this locally by running e.g., `VERSION=9.9.9-beta.9 make image chart` and examining the resulting artifacts (though the working tree will also be dirty):

```
$ helm show chart packages/cray-sts-9.9.9-beta.9.tgz 
annotations:
  artifacthub.io/changes: |
    - kind: security
      description: Updated images to CSM patched versions
      links:
        - name: Github Issue
          url: https://github.com/Cray-HPE/cray-sts/issues/12
        - name: Github PR
          url: https://github.com/Cray-HPE/cray-sts/pull/11
  artifacthub.io/images: |
    - name: cray-sts
      image: artifactory.algol60.net/csm-docker/stable/cray-sts:9.9.9-beta.9
  artifacthub.io/license: MIT
apiVersion: v2
appVersion: 9.9.9-beta.9
dependencies:
- name: cray-service
  repository: https://artifactory.algol60.net/artifactory/csm-helm-charts/
  version: ~7.0.0
description: Kubernetes resources for cray-sts
home: https://github.com/Cray-HPE/cray-sts
keywords:
- cray-sts
maintainers:
- name: kimjensen-hpe
- name: bklei
- name: brantk-hpe
name: cray-sts
version: 9.9.9-beta.9
```

The expectation is that the chart metadata should always be current, therefore it is recommended to run `make chart-metadata` whenever you change `.version` to automatically update the chart metadata (e.g., `version`, `appVersion`) as well as `global` settings in values.yaml and then commit the results.

File: Jenkinsfile.github (hunk 2 above)
Comment (zcrisler) [reply Jenkinsfile.github]: Shouldn't cause a problem, but fixed nonetheless.

**Review by zcrisler — APPROVED** [review]

**Pull request merged**